src/renderer/cursor_renderer/blink.rs

```diff
@@ -82,6 +83,32 @@ impl BlinkStatus {
         }
     }
 
+    /// Calculate the opacity the cursor should be drawn with when smooth cursor blink is enabled.
+    /// `0.0` is fully transparent, `1.0` is fully opaque.
+    pub fn opacity(&self) -> f32 {
+        let now = Instant::now();
+        if self.state == BlinkState::Waiting {
+            return 1.0;
+        }
+        let total = self.get_delay().as_secs_f32();
```

Comment:
(might be a problem here with f32 if it's running for 2 weeks or more, the game Celeste has issues with that — I need to test around a bit locally though. ignore this for now)

Reply:
`get_delay` will never return more than `max(blinkwait, blinkon, blinkoff)` in milliseconds, so I don't think that concern applies in this case.

Reply:
Sounds solid then! Nevermind!